desc/equilibrium/coords.py

```diff
@@ -82,17 +86,27 @@ def map_coordinates(  # noqa: C901
     period = np.asarray(period)
     coords = coords % period
 
-    # can't use AD or jit compile this because of grid stuff
+    params = get_params(inbasis + basis_derivs, eq)
+
+    @functools.partial(jit, static_argnums=1)
     def compute(y, basis):
-        data = eq.compute(basis, grid=Grid(y, sort=False))
+        grid = Grid(y, sort=False, jitable=True)
+        profiles = get_profiles(inbasis + basis_derivs, eq, grid, jitable=True)
+        # do surface average to get iota once
```

Comment:
maybe I dont understand this logic, it seems to me that if a current profile constrained eq is  used, it will get iota every time?

Reply:
by "once" mean per iteration, as opposed to the standard method which would do the averaging once for `iota` and once for `iota_r` etc.